lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,17 @@
+namespace :upgrade do
+  desc "Run sidekiq with old queues so it can finish deferred jobs"
+  task :run_legacy_queues do
+    system "bundle exec sidekiq --daemon -q maintenance -q dispatch -q delete_account -q http -q http_service"\
+           " -q export -q photos -q socket_webfinger -q mail -q receive_local -q receive -q receive_salmon"\
```

Comment:
I pushed updated version, it still daemonizes, but it can check whether queues are empty and run only queues with jobs. If we don't want daemonizing we could tell podmins to check whether there are non-empty queues and keep the legacy sidekiq process.